tests/lava/proc/event_data/event_data_loader/test_aedat_data_loader.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+import unittest
+from lava.proc.event_data.event_data_loader.aedat_data_loader import AedatDataLoader, \
+    AedatDataLoaderPM
+from dv import AedatFile
+from dv.AedatFile import _AedatFileEventNumpyPacketIterator
+
+import typing as ty
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort, OutPort
+from lava.magma.core.process.variable import Var
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.run_configs import Loihi1SimCfg
+
+
+class RecvSparse(AbstractProcess):
+    def __init__(self,
+                 shape: tuple) -> None:
+        super().__init__(shape=shape)
+
+        self.in_port = InPort(shape=shape)
+
+        self.data = Var(shape=shape, init=np.zeros(shape, dtype=int))
+        self.idx = Var(shape=shape, init=np.zeros(shape, dtype=int))
+
+
+@implements(proc=RecvSparse, protocol=LoihiProtocol)
+@requires(CPU)
+class PyRecvSparsePM(PyLoihiProcessModel):
+    in_port: PyInPort = LavaPyType(PyInPort.VEC_SPARSE, int)
+
+    data: np.ndarray = LavaPyType(np.ndarray, int)
+    idx: np.ndarray = LavaPyType(np.ndarray, int)
+
+    def run_spk(self) -> None:
+        data, idx = self.in_port.recv()
+
+        self.data = np.pad(data,
+                           pad_width=(0, self.in_port.shape[0] - data.shape[0]))
+        self.idx = np.pad(idx,
+                          pad_width=(0, self.in_port.shape[0] - data.shape[0]))
+
+
+class TestProcessAedatDataLoader(unittest.TestCase):
+    def test_init(self):
+        """Tests instantiation of AedatDataLoader"""
+        data_loader = AedatDataLoader(file_path="../dvs_recording.aedat4",
+                                      shape_out=(43200,))
+
+        self.assertIsInstance(data_loader, AedatDataLoader)
+        self.assertEqual(data_loader.proc_params["file_path"],
+                         "../dvs_recording.aedat4")
+        self.assertEqual(data_loader.proc_params["shape_out"], (43200,))
+
+    def test_invalid_file_path_extension(self):
+        """Tests for invalid file extension"""
+        with(self.assertRaises(ValueError)):
+            # Test for .py
+            _ = AedatDataLoader(file_path="test_aedat_data_loader.py",
```

Comment:
You can remove the `_ = ` before the `AedatDataLoader(...)`.

Reply:
done, removed for all tests